### PR TITLE
Bump machine-controller to v1.41.0

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
**What this PR does / why we need it**:
Bump machine controller in KKP from `v1.36.2` to `v1.41.0`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
Regarding the spellcheck exceptions, these misspelled words are coming from `kubevirt.io/api/core/v1/schema_swagger_generated.go` and temporarily we have to add them to ignore list, till it's fixed in the upstream source.

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
machine-controller is updated to v1.41.0
```

### CODE REVIEW

**Comment by ahmedwaleedmalik** [comment]
/hold
Before we move ahead with this PR, we need a new minor release for machine-controller `v1.41.0`.

File: cmd/kubermatic-api/swagger.json [hunk 1]
```diff
@@ -19562,7 +19562,7 @@
       "x-go-package": "k8c.io/kubermatic/v2/pkg/crd/kubermatic/v1"
     },
     "ClientPassthroughDevices": {
-      "description": "The struct is currently empty as there is no immediate request for\nuser-facing APIs. This structure simply turns on USB redirection of\nUsbClientPassthroughMaxNumberOf devices.",
+      "description": "The struct is currently empty as there is no imediate request for\nuser-facing APIs. This structure simply turns on USB redirection of\nUsbClientPassthroughMaxNumberOf devices.",
```
Comment (moadqassem) [inline cmd/kubermatic-api/swagger.json:19565]: typo: 
`s/imediate/immediate`

File: cmd/kubermatic-api/swagger.json [hunk 2]
```diff
@@ -25412,7 +25412,7 @@
     "Port": {
       "description": "Default protocol TCP.\nThe port field is mandatory",
       "type": "object",
-      "title": "Port represents a port to expose from the virtual machine.",
+      "title": "Port repesents a port to expose from the virtual machine.",
```
Comment (moadqassem) [inline cmd/kubermatic-api/swagger.json:25415]: typo:
`s/repesents/represents`

**Review by moadqassem — COMMENTED** [review]
couple of typos, but in general looks good

**Pull request merged**